keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/SecurityGroupHandler.kt

```diff
@@ -190,6 +178,68 @@ class SecurityGroupHandler(
       }
     }
 
+  private fun SecurityGroup.toCreateJob(): Job =
+    Job(
+      "upsertSecurityGroup",
+      mapOf(
+        "application" to application,
+        "credentials" to accountName,
+        "cloudProvider" to CLOUD_PROVIDER,
+        "name" to name,
+        "regions" to listOf(region),
+        "vpcId" to cloudDriverCache.networkBy(vpcName, accountName, region).id,
+        "description" to description,
+        "securityGroupIngress" to inboundRules
+          // we have to do a 2-phase create for self-referencing ingress rules as the referenced
+          // security group must exist prior to the rule being applied. We filter then out here and
+          // the subsequent diff will apply the additional group(s).
+          .filterNot { it is SelfReferenceRule }
```

Comment:
Filter out self-referencing rules when submitting the task to create a new security group to Orca